code/modules/power/singularity/collector.dm

```diff
@@ -163,3 +163,85 @@ var/global/list/rad_collectors = list()
 		flick("ca_deactive", src)
 	update_icons()
 	return
+
+/obj/machinery/power/rad_collector/pipenet
+	name = "Modular Radiation Collector Array"
+	desc = "A variant of the radiation collector array which connects to a pipecap underneath it."
+	var/obj/machinery/atmospherics/pipe/cap/pipe = null
+	var/datum/gas_mixture/removed = null
+
```

Comment:
You went a little bit overkill and removed the connecting to a pipenet in New() part. Also you can just stuff it in Initialize instead.